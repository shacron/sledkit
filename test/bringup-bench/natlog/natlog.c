#include "libmin.h"

/* calculate e=2.718..., using an interative approximation */

int
main(void)
{
  /* STEPS is usually a very large number eg 10000000 */
  int steps = 100000;
  double x, y;

  y = 1.0 + 1.0/steps;
  x = 1.0;

  for(; steps > 0; steps--)
    x *= y;

  libmin_printf("natlog: e=%f\n", x);

  libmin_success();
  return 0;
}

