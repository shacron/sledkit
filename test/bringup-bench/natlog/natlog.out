natlog: e=2.718268
